docs/customruns.md

```diff
@@ -0,0 +1,413 @@
+<!--
+---
+linkTitle: "CustomRuns"
+weight: 800
+---
+-->
+
+# CustomRuns
+
+- [Overview](#overview)
+- [Configuring a `CustomRun`](#configuring-a-customrun)
+  - [Specifying the target Custom Task](#specifying-the-target-custom-task)
+  - [Cancellation](#cancellation)
+  - [Specifying `Timeout`](#specifying-timeout)
+  - [Specifying `Retries`](#specifying-retries)
+  - [Specifying Parameters](#specifying-parameters)
+  - [Specifying Workspaces](#specifying-workspaces)
+  - [Specifying Service Account](#specifying-a-serviceaccount)
+- [Monitoring execution status](#monitoring-execution-status)
+  - [Monitoring `Results`](#monitoring-results)
+- [Code examples](#code-examples)
+  - [Example `CustomRun` with a referenced custom task](#example-customrun-with-a-referenced-custom-task)
+  - [Example `CustomRun` with an unnamed custom task](#example-customrun-with-an-unnamed-custom-task)
+  - [Example of specifying parameters](#example-of-specifying-parameters)
+
+# Overview
+
+A `CustomRun` allows you to instantiate and execute a [Custom
+Task](https://github.com/tektoncd/community/blob/main/teps/0002-custom-tasks.md),
+which can be implemented by a custom task controller running on-cluster. Custom
+Tasks can implement behavior that doesn't correspond directly to running a
+workload in a `Pod` on the cluster. For Pod-based on-cluster workloads, you
```

Comment:
Is there any way we can reword these guidelines around when to use a TaskRun vs CustomRun that don't focus on implementation details? For example, in the buildkit implementation of Tekton, TaskRuns don't run in a pod. (Maybe focusing on containers would be OK though.)

Reply:
SG!